ruptures/costs/costrank.py

```diff
@@ -97,23 +97,21 @@ def fit(self, signal):
 
         obs, vars = signal.shape
 
-        # argsort gives us 0-based ranks in the range [0, n-1]. Add one to put it in the range [1, n]
+        # argsort gives us 0-based ranks in the range [0, n-1].
+        # Add one to put it in the range [1, n]
         ranks = np.argsort(signal, axis=0) + 1
         # Center the ranks into the range [-(n+1)/2, (n+1)/2]
         centered_ranks = (ranks - ((obs + 1) / 2)).astype(int)
         # Sigma is the covariance of these ranks.
         # If it's a scalar, reshape it into a 1x1 matrix
         cov = np.cov(centered_ranks, rowvar=False, bias=True).reshape(vars, vars)
 
-        # Possibly the inversion could be done more efficiently with the knowledge that
-        # the cov matrix is positive semidefinite
-        self.inv_cov = inv(cov)
+        # Use the pseudoinverse if we have linear dependency
+        # see Lung-Yut-Fong, A., Lévy-Leduc, C., & Cappé, O. (2015)
         try:
             self.inv_cov = inv(cov)
```

Comment:
We could compute the pseudo-inverse matrix directly (`inv` and `pinv` coincide when `cov` is invertible) and keep the try-catch block in case the SVD (in pinv) does not converge:
```
try:
    self.inv_cov = pinv(cov)
except LinAlgError as e:
            raise LinAlgError(
                "The covariance matrix of the rank signal is not invertible and the pseudo-inverse computation did not converge."
            ) from e
```